### PR TITLE
Community: integrate chat models with Yuan2.0

### PR DESCRIPTION
**Pull request by cauwulixuan**
1. integrate chat models with [`Yuan2.0`](https://github.com/IEIT-Yuan/Yuan-2.0/blob/main/README-EN.md)
2. add a new doc for [Yuan2.0 integration](docs/docs/integrations/llms/yuan2.ipynb)
 
Yuan2.0 is a new generation Fundamental Large Language Model developed by IEIT System. We have published all three models, Yuan 2.0-102B, Yuan 2.0-51B, and Yuan 2.0-2B.

### CODE REVIEW

File: libs/community/langchain_community/chat_models/yuan2.py [hunk 1]
```diff
@@ -0,0 +1,506 @@
+"""ChatYuan2 wrapper."""
+from __future__ import annotations
+
+import asyncio
+import logging
+import os
+
+os.environ["no_proxy"] = "localhost,127.0.0.1,::1"
```
Comment (baskaryan) [inline libs/community/langchain_community/chat_models/yuan2.py:8]: delete

Comment (cauwulixuan) [reply libs/community/langchain_community/chat_models/yuan2.py:8]: Sure, I'll fix this, Thanks.

Comment (cauwulixuan) [reply libs/community/langchain_community/chat_models/yuan2.py]: @baskaryan I've deleted unnecessary import and related code block, it still stuck in deploying, did I miss anything? Thanks, PTAL :)

**Comment by cauwulixuan** [comment]
I've deleted unnecessary import block and re-format related code block, it still failed in deployment, Can you give me some tips how to fix this? Thanks. @baskaryan @hwchase17

**Comment by cauwulixuan** [comment]
I've fix some lint error.  And I can see checks failed with:
```
    from langchain_community.chat_models.yuan2 import ChatYuan2
  File "/home/runner/work/langchain/langchain/libs/community/langchain_community/chat_models/yuan2.py", line 55, in <module>
    from openai.types.chat import (
ModuleNotFoundError: No module named 'openai'
```
But I'm not familier with `poetry`,  my code was depended on `openai` package, but it seems that this package was not installed while check was running. How can I fix this? 

Any suggestions would be appreciate. Thanks @baskaryan @hwchase17

**Comment by cauwulixuan** [comment]
Any suggestions or any updates here? @baskaryan @hwchase17

**Review by baskaryan — APPROVED** [review]
thanks @cauwulixuan!

**Pull request merged**